### PR TITLE
Split Windows setup scripts, prepare for cleanup and multiple CRI

### PR DESCRIPTION
**Pull request by PatrickLang**
**What this PR does / why we need it**:
The Windows setup scripts were difficult to manage for a few reasons:
- Although the files looked like PowerShell scripts, they were subject to some limitations:
  - Golang was used to double-escape some characters so it could be put into ARM customData, and some like '`' couldn't be used
- Testing was difficult
  - The single script was interpreted as golang templates, which means you can't copy a file from the repo to a machine to test it
  - The scripts were not idempotent

This changes things so that only `kuberneteswindowssetup.ps1` is used as a golang template. All changes that are done by acs-engine are in that single file. They are validated and passed as parameters to powershell functions in separate files now. To avoid the escaping problems - all of those files are enclosed in a ZIP, base64 encoded, and extracted at runtime.

If you need to test changes, you can now:
- go change or replace c:\azuredata\k8s\*.ps1
- re-run c:\azuredata\customdatasetupscript.ps1 with the same parameters

This is the first PR towards https://github.com/Azure/acs-engine/issues/3851

### CODE REVIEW

**Comment by PatrickLang** [comment]
/hold
I'm still doing more testing on this, but want to take advantage of automation and start reviews.

**Comment by PatrickLang** [comment]
scheduled extra tests using kubenet/wincni. will remove hold after those pass

**Comment by PatrickLang** [comment]
/remove hold
kubenet tests passed 8/10. Since those are low priority and not blocking, I think that's good enough.

File: pkg/acsengine/engine.go [hunk 1]
```diff
@@ -650,8 +650,8 @@ func getSecurityRules(ports []int) string {
 	return buf.String()
 }
 
-// getSingleLineForTemplate returns the file as a single line for embedding in an arm template
-func (t *TemplateGenerator) getSingleLineForTemplate(textFilename string, cs *api.ContainerService, profile interface{}) (string, error) {
+// getSingleLineForTemplate returns the file as a single line
```
Comment (jackfrancis) [inline pkg/acsengine/engine.go]: method comment copy/paste cruft (should be `// getSingleLine ...`)

**Comment by jackfrancis** [comment]
we'll want to fix the comment nit to avoid lint failures in the future, otherwise lgtm

**Pull request merged**